superset-frontend/plugins/plugin-chart-echarts/src/BigNumber/BigNumberWithTrendline/transformProps.ts

```diff
@@ -107,13 +108,54 @@ export default function transformProps(
       // sort in time descending order
       .sort((a, b) => (a[0] !== null && b[0] !== null ? b[0] - a[0] : 0));
 
-    bigNumber = sortedData[0][1];
     timestamp = sortedData[0][0];
 
-    if (bigNumber === null) {
-      bigNumberFallback = sortedData.find(d => d[1] !== null);
-      bigNumber = bigNumberFallback ? bigNumberFallback[1] : null;
-      timestamp = bigNumberFallback ? bigNumberFallback[0] : null;
+    const metricValues: number[] = sortedData
+      .map(d => d[1])
+      .filter((value): value is number => value !== null);
+
+    if (metricValues.length > 0) {
+      const sortedValues: number[] = [...metricValues].sort((a, b) => a - b);
+      const totalSum: number = metricValues.reduce(
+        (sum, value) => sum + value,
+        0,
+      );
+      const mid = Math.floor(sortedValues.length / 2);
```

Comment:
It's better to move this inside `case 'MEDIAN':`, so it's only computed if needed.